### PR TITLE
"[HTP-0001]  AdYouLike adaptor : create adaptor

### PR DESCRIPTION
**Pull request by guiann**
## Type of Change
- [x] New adapter
- [ ] Feature Update
- [ ] Bug Fix
- [ ] Code Refactor
- [ ] Other

## Description of Change
Create new AdYouLike bidding adapter 

Please note that the link to guideline in default PR description is outdated:
https://knowledgebase.indexexchange.com/display/ADAPTER/Adapter+Code+Submission+Guidelines

### CODE REVIEW

**Comment by guiann** [comment]
This[ documentation page](https://kb.indexexchange.com/ix_library_partners/build_adapter/log_a_ticket_with_IX_to_request_certification.htm) says I have to merge the code into the canary branch but "**Only those with write access to this repository can merge pull requests**."

please advice.

**Comment by guiann** [comment]
> This[ documentation page](https://kb.indexexchange.com/ix_library_partners/build_adapter/log_a_ticket_with_IX_to_request_certification.htm) says I have to merge the code into the canary branch but "**Only those with write access to this repository can merge pull requests**."
> 
> please advice.

Answer from Chelsee Donovan via e-mail:

"Confirming I see Adyoulike’s pull request in GitHub – you are all set. You will not need to merge into canary, IX will handle this for you once we have a chance to review the code."

File: ad-you-like/ad-you-like-htb.js [hunk 1]
```diff
@@ -0,0 +1,576 @@
...
+            if (!adserverOrgIp && bid.identityData) {
+                adserverOrgIp = bid.identityData.AdserverOrgIp;
+            }
+
+            return accumulator;
+        }, {});
+
+        /* Change this to your bidder endpoint. */
+        var baseUrl = Browser.getProtocol() + '//hb-api.omnitagjs.com/hb-api/ix/v1?';
+        baseUrl += 'RefererUrl=' + encodeURIComponent(Browser.getReferrer());
```
Comment (ix-certification) [inline ad-you-like/ad-you-like-htb.js:192]: Include query params as part of the query object , 
You can leave the baseUrl without the query params 
i.e baseUrl should just be  `Browser.getProtocol() + '//hb-api.omnitagjs.com/hb-api/ix/v1`;
Any query params you want to add should be part of the query object. 

This is causing your system tests to fail, so you can verify your fix by running the system test ... 

https://kb.indexexchange.com/ix_library_partners/build_adapter/run_the_adapter_test_cases.htm

Comment (guiann) [reply ad-you-like/ad-you-like-htb.js:192]: Will do.  Thanks

Comment (guiann) [reply ad-you-like/ad-you-like-htb.js]: I've made the change but every try to run the debug script will result in an error from the adapter-development-suite.
Seems to happen on other adapters too.


`> ht-wrapper-adapters@1.0.0 debug
> node ./node_modules/adapter-development-suite/bin/index.js

/home/user/path/ix-library-adapters/node_modules/adapter-development-suite/bin/index.js:86
    if (cwd.length - cwd.lastIndexOf('ht-wrapper-adapters') === repoNameLength
            ^

TypeError: Cannot read property 'length' of undefined
    at Object.<anonymous> (/home/user/path/ix-library-adapters/node_modules/adapter-development-suite/bin/index.js:86:13)
    at Module._compile (node:internal/modules/cjs/loader:1083:30)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1112:10)
    at Module.load (node:internal/modules/cjs/loader:948:32)
    at Function.Module._load (node:internal/modules/cjs/loader:789:14)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:72:12)
    at node:internal/main/run_main_module:17:47
npm ERR! code 1
npm ERR! path /home/user/path/ix-library-adapters
npm ERR! command failed
npm ERR! command sh -c node ./node_modules/adapter-development-suite/bin/index.js

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/guillaume/.npm/_logs/2021-02-04T16_18_56_047Z-debug.log
`

Comment (ix-certification) [reply ad-you-like/ad-you-like-htb.js]: Hi , which repo and folder are you running this from ?

```
ht-wrapper-adapters@1.0.0 debug
node ./node_modules/adapter-development-suite/bin/index.js
/home/user/path/ix-library-adapters/node_modules/adapter-development-suite/bin/index.js:86
if (cwd.length - cwd.lastIndexOf('ht-wrapper-adapters') === repoNameLength
^
TypeError: Cannot read property 'length' of undefined
```

Comment (guiann) [reply ad-you-like/ad-you-like-htb.js]: Hello IX team. 
We have been advised we would be included on your "next sprint". Can you please let us know when that will be ?
I'm still waiting for an answer to make sure everything is fine with our adapter?

Comment (ix-certification) [reply ad-you-like/ad-you-like-htb.js]: Hi Guiann 
Yes, this work has been included in this sprint and we are actively working on adding your adapter.

Comment (guiann) [reply ad-you-like/ad-you-like-htb.js]: Thank you,

when running  'npm run debug'  from our adapter folder I don't see any error anymore. And I'm able to display an ad in the localhost test page.

File: ad-you-like/CHANGES.md [hunk 2]
```diff
@@ -0,0 +1,4 @@
+# 1.0
+ 
+- Create AdouLike adapter
```
Comment (punkiller) [inline ad-you-like/CHANGES.md:3]: typo  Ad'Y'ouLike

File: ad-you-like/ad-you-like-htb.js (hunk 1 above)
Comment (ix-certification) [reply ad-you-like/ad-you-like-htb.js]: Yes confirming, no errors there .

File: ad-you-like/CHANGES.md (hunk 2 above)
Comment (ix-certification) [inline ad-you-like/CHANGES.md:3]: type in AdYouLike

File: ad-you-like/DOCUMENTATION.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
+# Ad You Like
+## General Compatibility
+|Feature|  |
+|---|---|
+| Consent |  |
+| Native Ad Support |  |
```
Comment (ix-certification) [inline ad-you-like/DOCUMENTATION.md:6]: Please try and fill out these tables here.
you can reference examples of other adapters, like  
https://github.com/Adyoulike/ix-library-adapters/blob/AdYouLike-adaptor/emx-digital/DOCUMENTATION.md

File: ad-you-like/CHANGES.md (hunk 2 above)
Comment (guiann) [reply ad-you-like/CHANGES.md]: solved

File: ad-you-like/DOCUMENTATION.md (hunk 3 above)
Comment (guiann) [reply ad-you-like/DOCUMENTATION.md]: Done

File: ad-you-like/ad-you-like-htb.js (hunk 1 above)
Comment (guiann) [reply ad-you-like/ad-you-like-htb.js]: Thank you

**Comment by guiann** [comment]
Changes have been made:  fix typoe and fill out documentation file.

**Pull request merged**